lib/ansible/modules/network/cloudengine/ce_vxlan_gateway.py

```diff
@@ -0,0 +1,947 @@
+                self.existing["arp_distribute_gateway"] = "enable"
+            else:
+                self.existing["arp_distribute_gateway"] = "disable"
+            if is_config_exist(self.config, "arp direct-route enable"):
+                self.existing["arp_direct_route"] = "enable"
+            else:
+                self.existing["arp_direct_route"] = "disable"
+
+    def get_end_state(self):
+        """get end state info"""
+
+        config = self.get_current_config()
+        if not config:
+            return
+
+        if is_config_exist(config, "dfs-group 1"):
+            self.end_state["dfs_id"] = "1"
+            self.end_state["dfs_source_ip"] = get_dfs_source_ip(config)
+            self.end_state["dfs_source_vpn"] = get_dfs_source_vpn(config)
+            self.end_state["dfs_udp_port"] = get_dfs_udp_port(config)
+            if is_config_exist(config, "active-active-gateway"):
+                self.end_state["dfs_all_active"] = "enable"
+                self.end_state["dfs_peers"] = get_dfs_peers(config)
+            else:
+                self.end_state["dfs_all_active"] = "disable"
+
+        if self.vpn_instance:
+            self.end_state["vpn_instance"] = get_ip_vpn(config)
+            self.end_state["vpn_vni"] = get_ip_vpn_vni(config)
+
+        if self.vbdif_name:
+            self.end_state["vbdif_name"] = self.vbdif_name
+            self.end_state["vbdif_mac"] = get_vbdif_mac(config)
+            self.end_state["vbdif_bind_vpn"] = get_vbdif_vpn(config)
+            if is_config_exist(config, "arp distribute-gateway enable"):
+                self.end_state["arp_distribute_gateway"] = "enable"
+            else:
+                self.end_state["arp_distribute_gateway"] = "disable"
+            if is_config_exist(config, "arp direct-route enable"):
+                self.end_state["arp_direct_route"] = "enable"
+            else:
+                self.end_state["arp_direct_route"] = "disable"
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        self.config = self.get_current_config()
+        self.get_existing()
+        self.get_proposed()
+
+        # deal present or absent
+        if self.dfs_id:
+            self.config_dfs_group()
+
+        if self.vpn_instance:
+            self.config_ip_vpn()
+
+        if self.vbdif_name:
+            self.config_vbdif()
+
+        if self.commands:
+            self.cli_load_config(self.commands)
+            self.changed = True
+
+        self.get_end_state()
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        if self.changed:
+            self.results['updates'] = self.updates_cmd
+        else:
+            self.results['updates'] = list()
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """Module main"""
+
+    argument_spec = dict(
```

Comment:
For integers please:
use `type='int`
remove the `isdigit` check from `check_params`